language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java

```diff
@@ -140,4 +156,42 @@ public static CodeActionNodeType topLevelNodeInLine(TextDocumentIdentifier ident
             return null;
         }
     }
+
+    /**
+     * Translates ballerina diagnostics into lsp4j diagnostics.
+     *
+     * @param ballerinaDiags a list of {@link org.ballerinalang.util.diagnostic.Diagnostic}
+     * @return a list of {@link Diagnostic}
+     */
+    public static List<org.eclipse.lsp4j.Diagnostic> toDiagnostics(
+            List<org.ballerinalang.util.diagnostic.Diagnostic> ballerinaDiags) {
+        List<org.eclipse.lsp4j.Diagnostic> lsDiagnostics = new ArrayList<>();
+        ballerinaDiags.forEach(diagnostic -> {
+            org.eclipse.lsp4j.Diagnostic lsDiagnostic = new org.eclipse.lsp4j.Diagnostic();
+            lsDiagnostic.setSeverity(DiagnosticSeverity.Error);
+            lsDiagnostic.setMessage(diagnostic.getMessage());
+            Range r = new Range();
+
+            int startLine = diagnostic.getPosition().getStartLine() - 1; // LSP diagnostics range is 0 based
+            int startChar = diagnostic.getPosition().getStartColumn() - 1;
+            int endLine = diagnostic.getPosition().getEndLine() - 1;
+            int endChar = diagnostic.getPosition().getEndColumn() - 1;
```

Comment:
Can we use https://github.com/ballerina-platform/ballerina-lang/blob/master/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java#L186 to convert the diagnostic position to zero-based positioning?

Reply:
Will fix in a separate PR